### ISSUE TITLE
Potential name clash between components and events/commands in generated code

### ISSUE THREAD

**Issue by jamiebrynes7** [spatialos/gdk-for-unity#957]
**Affects:** Release v0.1.0 and up 
**Internal Ticket:** [UTY-1962](https://improbableio.atlassian.net/browse/UTY-1962)

---

#### Description

Given a schema file like one of the following:

```
package name;

type Foo {
    float test_field = 1;
}

component Bar
{
    id = 200;
    event Foo bar;
}
```

```
package name;

type Foo {
    float test_field = 1;
}

component Bar
{
    id = 200;
    command Foo bar(Foo);
}
```

The code generator will generate invalid C# code that does not compile.

> Note this occurs when the name of a component and the name of an event/command inside that component are _the same_

#### Workaround

Rename either the component or the event/command.

**Comment by paulbalaji** [comment]
A resolution has been merged and will be part of our 0.3.1 release next week.

**Comment by jamiebrynes7** [comment]
Released in https://github.com/spatialos/gdk-for-unity/releases/tag/0.3.1
